tss-esapi/tests/integration_tests/error_tests.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+mod return_code_tests;
+mod wrapper_error_kind_tests;
+
+use std::{convert::TryFrom, error::Error};
+
+use tss_esapi::{
+    constants::tss::{TPM2_RC_INITIALIZE, TSS2_TPM_RC_LAYER},
+    error::{ReturnCode, WrapperErrorKind},
+};
+
+#[test]
+fn test_error_trait_implementation() {
+    // The Error type is only expected to forward everything to the
+    // underlaying error types.
+    let expected_wrapper_error_kind = WrapperErrorKind::InconsistentParams;
+    let wrapper_error = tss_esapi::Error::WrapperError(expected_wrapper_error_kind);
+    let actual_wrapper_error_kind = wrapper_error
+        .source()
+        .expect("`source()` for an Error of type WrapperError returned None.");
```

Comment:
I'm wondering whether it's useful to rewrite the expect message to use [the Recommended Message Style](https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style). :thinking:

Reply:
Fixed